ethcore/sync/src/chain/mod.rs

```diff
@@ -468,6 +485,20 @@ impl ChainSyncApi {
 		SyncSupplier::dispatch_packet(&self.sync, io, peer, packet_id, data)
 	}
 
+	/// Process the queue with requests, that were delayed with response.
+	pub fn process_delayed_requests(&self, io: &mut dyn SyncIo) {
+		let requests;
+		{
+			requests = self.sync.write().retrieve_delayed_requests();
+		}
```

Comment:
```suggestion
		let requests= self.sync.write().retrieve_delayed_requests();
```

This is fine, the lock guard will be dropped.